Scuti/Plugins/iOS/WebView.mm

```diff
@@ -410,6 +410,15 @@ - (void)webView:(WKWebView *)wkWebView decidePolicyForNavigationAction:(WKNaviga
         UnitySendMessage([gameObjectName UTF8String], "CallFromJS", [[url substringFromIndex:6] UTF8String]);
         decisionHandler(WKNavigationActionPolicyCancel);
         return;
+    } else if (((navigationAction.targetFrame != nil && navigationAction.targetFrame.isMainFrame) || 
+            (navigationAction.navigationType == WKNavigationTypeLinkActivated ))//&& (!navigationAction.targetFrame || !navigationAction.targetFrame.isMainFrame)))
+
+            && hookRegex != nil && [hookRegex firstMatchInString:url options:0 range:NSMakeRange(0, url.length)]) {
+
+        //NSLog(@"HOOKED %@ in  %s",   navigationAction, [gameObjectName UTF8String]); 
+         UnitySendMessage([gameObjectName UTF8String], "CallOnHooked", [url UTF8String]);
+            decisionHandler(WKNavigationActionPolicyCancel);
+            return; 
```

Comment:
@markgrossnickle Please, let me know if this is fine

Reply:
@juanesdavinci that looks fine but why is it duplicated below?

Reply:
That is another case, that was set by the update

Reply:
For me the conditions where different so I left it to fallback after the one you added

Reply:
I see, but this negates the prior check.  This check we are commenting on says if ((a || b) && c) then hook and you are adding elseif c then hook. Which then means we always hook on C and thus aren't checking for a || b. Those aren't needed.

But checking for a || b and hooking on those are needed. If we remove those then we hook all ads that try to load into iframes.  Some ads include the scuti url such as having ref=scuti.store.... blah blah in them so the regex checking for the scuti url gets triggered when loading images of ads in iframes and was popping up the secondary ad window when it shouldn't be. 

So we need these checks. What is the use case you are trying to solve here?   @juanesdavinci

Reply:
In that case, we don't need `c` in this conditional, right? just 

`else if (((navigationAction.targetFrame != nil && navigationAction.targetFrame.isMainFrame) || 
            (navigationAction.navigationType == WKNavigationTypeLinkActivated ))`

If I understand correctly, It just have to check for the href. 

I only copied and pasted what it was in the previous doc, but I might be mistaken

Reply:
I don't understand. @juanesdavinci 

We need the above check for ((a || b) && c).  I know we need that with certainty after testing ads endlessly with client andrew. 

In my view, we don't need the below. We can't hook everything that passes HookRegex. We can only hook things that pass the regex and are from a click or are targeting the main thread. Otherwise we get unwanted behavior. So remove the following check and we are good. If you think we need the following check then explain what case it fixes that the above does not already cover.

Reply:
Ok, I will remove it,